### PR TITLE
Implement ElementalType

### PR DESCRIPTION
**Pull request by krixstin**

### CODE REVIEW

File: element.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
+package pokemonbattlelib
+
+import "fmt" 
+type ElementalType uint32
+var effect int
+
+const (
+	Normal ElementalType = 65536 >> iota
```
Comment (dyc3) [inline element.go:8]: It'll be easier to work with if you do `1 << iota` instead

File: element.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      }
+   }
+   return false
+}
+ 
+func getEffect(move, def ElementalType) int {
+    var match =uint32(move| def)
+    fmt.Println("match spot: ", uint32(move| def))
+    fmt.Printf("%b", int32(132096))
+    noEffect := []uint32{132096,66560,16896,40960,133120,8208,9}
```
Comment (dyc3) [inline element.go:40]: You can enter binary literal numbers by starting the number with `0b`. For example: `0b101 == 5`

File: element.go (hunk 1 above)
Comment (krixstin) [reply element.go:8]: Wouldn't the chart be flipped?

File: element.go (hunk 2 above)
Comment (krixstin) [reply element.go:40]: Thank you. They are easier to read

File: elemental_type.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+ 
+func GetEffect(move, def ElementalType) (int,error) {
+
+    noEffect := map[ElementalType]ElementalType{
+        Normal: Ghost,
+        Fight: Ghost,
+        Poison: Steel,
+        Ground: Flying,
+        Ghost: Normal,
+        Electric: Rock,
```
Comment (carpenterd777) [inline elemental_type.go:35]:
```suggestion
        Electric: Ground,
```

File: elemental_type.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        Poison: Grass,
+        Ground: Poison| Rock| Steel| Fire| Electric,
+        Rock: Flying| Bug| Fire| Ice,
+        Bug: Grass |Psychic| Dark,
+        Ghost: Ghost| Psychic,
+        Steel: Rock| Ice,
+        Fire: Bug| Steel| Grass| Ice,
+        Water: Ground| Rock| Fire,
+        Grass: Ground| Rock| Water,
+        Electric: Fight|Poison,
```
Comment (carpenterd777) [inline elemental_type.go:71]:
```suggestion
        Electric: Flying | Poison,
```

File: elemental_type.go [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        Psychic: Steel  | Psychic,
```
Comment (carpenterd777) [inline elemental_type.go:72]:
```suggestion
        Psychic: Fight | Poison
```

**Review by carpenterd777 — CHANGES_REQUESTED** [review]
yeah looks good to me, you've just got a couple inaccuracies in the tables and it looks like you've got some merge conflicts to resolve

**Review by dyc3 — APPROVED** [review]

File: elemental_type.go [hunk 6]
```diff
@@ -1,5 +1,135 @@
...
+
+	reduce := bits.OnesCount32(uint32(halfEffect[move] & def))
+	increase := bits.OnesCount32(uint32(doubleEffect[move] & def))
+	effect := (increase - reduce) * 2
+	if effect == 0 {
+		return NormalEffect
+	}
+	if effect > 0 {
+		return Effectiveness(effect)
+	} else {
```
Comment (carpenterd777) [inline elemental_type.go:122]: (nit)
```suggestion
	} else if effect > 0 {
		return Effectiveness(effect)
	} else {
```

**Review by carpenterd777 — APPROVED** [review]

**Pull request merged**